src/vscode-dts/vscode.proposed.terminalShellType.d.ts

```diff
@@ -0,0 +1,39 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+declare module 'vscode' {
+
+	// https://github.com/microsoft/vscode/issues/230165
+
+	/**
+	 * Known terminal shell types.
+	 */
+	export enum TerminalShellType {
+		Sh = 1,
+		Bash,
+		Fish,
+		Csh,
+		Ksh,
+		Zsh,
+		CommandPrompt,
+		GitBash,
+		PowerShell,
+		Python,
+		Julia,
+		NuShell
```

Comment:
We could add Node as a shell type after updating the branch https://github.com/microsoft/vscode/pull/238127

Reply:
Good idea, I added them [here](https://github.com/microsoft/vscode/pull/238071/commits/9644871021f4b6b8b05888b6d95f9a97ca36f7fc)
Now